merlin/models/tf/blocks/interaction.py

```diff
@@ -234,4 +235,78 @@ def call(self, inputs: tf.Tensor, **kwargs) -> tf.Tensor:
         return 0.5 * tf.subtract(summed_square, squared_sum)
 
     def compute_output_shape(self, input_shapes):
+        if len(input_shapes) != 3:
+            raise ValueError("Found shape {} without 3 dimensions".format(input_shapes))
         return (input_shapes[0], input_shapes[2])
+
+
+def FMBlock(
+    schema: Schema,
+    fm_input_block: Optional[Block] = None,
+    wide_input_block: Optional[Block] = None,
+    factors_dim: Optional[int] = None,
+    **kwargs,
+) -> tf.Tensor:
+    """Implements the Factorization Machine, as introduced in [1].
+    It consists in the sum of a wide component that weights each
+    feature individually and a 2nd-level feature interaction using
+    factors.
+
+    References
+    ----------
+    [1] Steffen, Rendle, "Factorization Machines" IEEE International
+    Conference on Data Mining, 2010. https://ieeexplore.ieee.org/document/5694074
+
+    Parameters
+    ----------
+    schema : Schema
+        The schema of input features
+    fm_input_block : Optional[Block], by default None
+        The input block for the 2nd-order feature interaction in Factorization Machine.
+        Only categorical features will be used by this block, as it computes
+        dot product between all paired combinations of embedding values.
+        If not provided, an InputBlockV2 is instantiated based on schema.
+        Note: All features (including continuous) are considered in the
+        1st-order (wide) part which uses another input block.
+    wide_input_block: Optional[Block], by default None
+        The input for the wide block. If not provided,
+        creates a default block that encodes categorical features
+        with one-hot / multi-hot representation and also includes the continuous features.
+    factors_dim : Optional[int], optional
+        If fm_input_block is not provided, the factors_dim is used to define the
+        embeddings dim to instantiate InputBlockV2, by default None
+    Returns
+    -------
+    tf.Tensor
+        Returns a 2D tensor (batch size, 1) with the sum of the wide component
+        and 2nd-order interaction component of FM
+    """
+
+    cat_schema = schema.select_by_tag(Tags.CATEGORICAL)
+    cont_schema = schema.select_by_tag(Tags.CONTINUOUS)
+
+    wide_input_block = wide_input_block or ParallelBlock(
+        {
+            "categorical": CategoryEncoding(cat_schema, output_mode="multi_hot", sparse=True),
+            "continuous": SequentialBlock(Filter(cont_schema), ToSparseFeatures()),
+        },
+        aggregation="concat",
+    )
+
+    first_order = wide_input_block.connect(
+        MLPBlock([1], activation="linear", use_bias=True, **kwargs)
```

Comment:
Seems like `**kwargs` is being forwarded to a few different blocks, if possible we’d like to remove this as much as possible since it leads to more understandable code. If you want to allow for full customization is would be cleaner if a param like `wide_logit` be added to `FMBlock` and make `Dense(1, activation="linear", use_bias=True)` the default. MLPBlock is not needed since we only need one dense-layer.

Reply:
Makes sense. I was forwarding those **kwargs to be able to add regularization (dropout, kernel reg) to the wide part. But have created the `wide_logit_block` argument you suggested to make it more explicit.